src/Bedrock.Framework/Protocols/ProtocolWriter.cs

```diff
@@ -72,8 +72,10 @@ public async ValueTask WriteManyAsync<TWriteMessage>(IMessageWriter<TWriteMessag
                     return;
                 }
 
-                foreach (var protocolMessage in protocolMessages)
+                var enumerator = protocolMessages.GetEnumerator();
```

Comment:
Undo

Reply:
What even did I do here? Definitely a typo when putting this code back.